packages/debugger/src/panels/kernelSources/filter.tsx

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { InputGroup, ReactWidget } from '@jupyterlab/ui-components';
+
+import React, { useState } from 'react';
+
+import { IDebugger } from '../../tokens';
+
+/**
+ * The class name added to the filebrowser crumbs node.
+ */
+export interface IFilterBoxProps {
+  model: IDebugger.Model.IKernelSources;
+  filter: string;
+}
+
+const FilterBox = (props: IFilterBoxProps) => {
```

Comment:
The source filter truth should only be in one place (and preferably in the model).

It would better to define the filter box with props: `filter` and `onFilterChange`. And create a ReactWidget like:

```ts
<UseSignal signal={props.model.changed} initArgs={model.filter}>
    {(model) => <FilterBox filter={model.filter} onFilterChange={(e) => {model.filter = (e.target as HTMLInputElement).value}} />}
</UseSignal>
```

Reply:
Done. Thx!